### PR TITLE
Add message field to change

### PR DESCRIPTION
**Pull request by halink0803**

### CODE REVIEW

File: reservesetting/storage/postgres/sql_schema.go [hunk 1]
```diff
@@ -188,8 +191,9 @@ CREATE TABLE IF NOT EXISTS setting_change
 (
     id      SERIAL PRIMARY KEY,
     created TIMESTAMPTZ                 NOT NULL,
-    cat     setting_change_cat UNIQUE NOT NULL,
-    data    JSON                      NOT NULL
+    cat     setting_change_cat NOT NULL,
+    data    JSON                      NOT NULL,
+    status  setting_change_status NOT NULL DEFAULT 'pending'
```
Comment (secmask) [inline reservesetting/storage/postgres/sql_schema.go:196]: using `DEFAULT` with `NOT NULL` is weird?

File: reservesetting/storage/postgres/sql.go [hunk 2]
```diff
@@ -642,8 +642,8 @@ func settingChangeStatements(db *sqlx.DB) (*sqlx.Stmt, *sqlx.Stmt, *sqlx.Stmt, e
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	const deleteSettingChangeQuery = `DELETE FROM setting_change WHERE id=$1 returning id`
-	deleteSettingChangeStmt, err := db.Preparex(deleteSettingChangeQuery)
+	const updateSettingChangeStatus = `UPDATE setting_change SET status = $2 WHERE id=$1 returning id`
+	updateSettingChangeStatusStmt, err := db.Preparex(updateSettingChangeStatus)
 	if err != nil {
 		return nil, nil, nil, err
 	}
```
Comment (secmask) [inline reservesetting/storage/postgres/sql.go:649]: we should allow to filter by status too, default should be pending

Comment (halink0803) [reply reservesetting/storage/postgres/sql.go:649]: I have added filter option for api and storage query.

**Comment by secmask** [comment]
as Gin MR get merged, it's needed to resolve a bit

File: reservesetting/http/setting_change.go [hunk 3]
```diff
@@ -234,13 +234,23 @@ func (s *Server) getSettingChange(c *gin.Context) {
 	}
 	httputil.ResponseSuccess(c, httputil.WithData(result))
 }
+
+type getSettingChangeWitTypeFilter struct {
+	status common.ChangeStatus `form:"status"`
```
Comment (secmask) [inline reservesetting/http/setting_change.go:239]: private fill can not be filled

File: reservesetting/http/setting_change.go [hunk 4]
```diff
@@ -234,13 +234,23 @@ func (s *Server) getSettingChange(c *gin.Context) {
...
+
 func (s *Server) getSettingChangeWithType(t common.ChangeCatalog) func(ctx *gin.Context) {
 	return func(ctx *gin.Context) {
 		s.getSettingChanges(ctx, t)
 	}
 }
 func (s *Server) getSettingChanges(c *gin.Context, t common.ChangeCatalog) {
-	result, err := s.storage.GetSettingChanges(t)
+	var query getSettingChangeWitTypeFilter
+	if err := c.ShouldBindQuery(&query); err != nil {
```
Comment (secmask) [inline reservesetting/http/setting_change.go:249]: in case status param not set, will this return error or make pending as default?

Comment (halink0803) [reply reservesetting/http/setting_change.go:249]: as many old test does not have status param set, it make sure the bind query will return pending as default: https://github.com/KyberNetwork/reserve-data/blob/b034aaa40157160ce7eab1be5ae9d2532b743510/reservesetting/http/setting_change_test.go#L881

File: reservesetting/http/setting_change.go (hunk 3 above)
Comment (halink0803) [reply reservesetting/http/setting_change.go]: update it as public.

File: reservesetting/http/setting_change.go (hunk 4 above)
Comment (halink0803) [reply reservesetting/http/setting_change.go:252]: I have update the default value

**Pull request merged**